src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -1903,18 +1904,21 @@ internal static bool CrackStringArrayInAttributeValue(out ImmutableArray<string?
             if (sig.RemainingBytes >= 4)
             {
                 uint arrayLen = sig.ReadUInt32();
-                var stringArray = new string?[arrayLen];
-                for (int i = 0; i < arrayLen; i++)
+                if (sig.RemainingBytes >= arrayLen)
```

Comment:
A null constant in metadata will return `0xFFFFFFFF` for `arrayLen` here. This is the same approach used by the other 2 array crackers in PEModule, `CrackByteArrayInAttributeValue` and `CrackBoolArrayInAttributeValue`.